### PR TITLE
New Local Notification Sample port from Legacy

### PR DESCRIPTION
**Pull request by officialgio**
Local Notification should allow the user to click on a button and the user should receive a notification of the times the button is clicked.

### CODE REVIEW

File: LocalNotifications/LocalNotifications/SecondActivity.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{
+    protected override void OnCreate (Bundle? bundle)
+    {
+        base.OnCreate (bundle);
+
+        // Get the count value passed to us from MainActivity:
+        var count = Intent?.Extras?.GetInt (MainActivity.COUNT_KEY, -1);
+
+        // No count was passed? Then just return.
+        if (count <= 0)
```
Comment (tj-devel709) [inline LocalNotifications/LocalNotifications/SecondActivity.cs:17]: Does this have any nullability issues?

File: LocalNotifications/LocalNotifications/MainActivity.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        base.OnCreate (bundle);
+        SetContentView (Resource.Layout.Main);
+
+        CreateNotificationChannel ();
+
+        // Display the "Hello World, Click Me!" button and register its event handler:
+        var button = FindViewById<Button> (Resource.Id.MyButton);
+        ArgumentNullException.ThrowIfNull (button);
+        button.Click += ButtonOnClick;
+
```
Comment (tj-devel709) [inline LocalNotifications/LocalNotifications/MainActivity.cs:33]:
```suggestion
```

**Review by tj-devel709 — CHANGES_REQUESTED** [review]
Great Start! Few suggestions and questions

File: LocalNotifications/LocalNotifications/MainActivity.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        // Increment the button press count:
+        count++;
+    }
+
+    void CreateNotificationChannel ()
+    {
+        ArgumentNullException.ThrowIfNull(Resources);
+
+        // API 26 and below works but are new in API 26+
```
[the remark this answers is not in the document]
Comment (officialgio) [reply LocalNotifications/LocalNotifications/MainActivity.cs]: I've modified the comment.

File: LocalNotifications/LocalNotifications/MainActivity.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        var notificationManager = NotificationManagerCompat.From (this);
+        notificationManager.Notify (NOTIFICATION_ID, builder.Build ());
+
+        // Increment the button press count:
+        count++;
+    }
+
+    void CreateNotificationChannel ()
+    {
+        ArgumentNullException.ThrowIfNull(Resources);
```
[the remark this answers is not in the document]
Comment (officialgio) [reply LocalNotifications/LocalNotifications/MainActivity.cs]: Done!

File: LocalNotifications/LocalNotifications/MainActivity.cs (hunk 2 above)
Comment (officialgio) [reply LocalNotifications/LocalNotifications/MainActivity.cs]: Done!

**Review by rachelkang — COMMENTED** [review]
I wasn't able to get the notifications working on my device with API 32, but apparently it's working fine on API 31. @jonathanpeppers do you have any ideas?

**Comment by jonathanpeppers** [comment]
@rachelkang do we need to add this:

https://developer.android.com/about/versions/13/changes/notification-permission

```xml
<uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
```

**Review by jonathanpeppers — APPROVED** [review]

**Pull request merged**